docs/src/model_setup/boundary_conditions.md

```diff
@@ -1,62 +1,177 @@
-   Tracer boundary conditions are no flux by default in `Bounded` directions.
-   Momentum boundary conditions are free-slip for tangential components and impenetrable for wall-normal components in `Bounded` directions.
-   
-2. Another way to say point 1 is that you'll never need to set:
-    * `Periodic` boundary conditions (default for `Periodic` directions);
-    * Impenetrable / "no normal flow" boundary conditions (default for wall-normal momentum components in `Bounded` directions);
-    * "No flux" or "free slip" boundary conditions (default for tracers and wall-tangential momentum components in `Bounded` directions).
+   Some applications of `ValueBoundaryCondition` are:
 
-3. `ValueBoundaryCondition` (aka "Dirichlet" boundary conditions) models boundary fluxes given a field's diffusive flux model, and assuming that a field has the prescribed value on the boundary.
-   _Note_: You cannot use `ValueBoundaryCondition` on a wall-normal velocity component; you must use `Open` for that.
-   Examples where you might use `ValueBoundaryCondition`:
-   * Prescribe a surface to have a constant temperature, like 20 degrees.
-     Heat will then flux in and out of the domain depending on the temperature difference between the surface and the interior, and the temperature diffusivity.
-   * Prescribe a velocity tangent to a boundary as in a driven-cavity flow (for example), where the top boundary is moving.
-     Momentum will flux into the domain do the difference between the top boundary velocity and the interior velocity, and the prescribed viscosity.
+       * no-slip boundary condition for wall-tangential velocity components via `ValueBoundaryCondition(0)`;
+       * surface temperature distribution, where heat fluxes in and out of the domain
+         at a rate controlled by the near-surface temperature gradient and the temperature diffusivity;
+       * constant velocity tangential to a boundary as in a driven-cavity flow (for example), 
+         where the top boundary is moving. Momentum will flux into the domain do the difference
+         between the top boundary velocity and the interior velocity, and the prescribed viscosity.
 
-4. `FluxBoundaryCondition` _directly_ prescribes the flux of a quantity across a boundary rather than calculating it given a viscosity or diffusivity.
-   For example, sunlight absorbed at the ocean surface imparts a temperature flux that heats near-surface fluid.
-   If there is a known `diffusivity`, you can express `FluxBoundaryCondition(flux)` using `GradientBoundaryCondition(-flux / diffusivity)` (aka "Neumann" boundary condition).
-   But when `diffusivity` is not known or is variable (as for large eddy simulation, for example), it's convenient and more straightforward to apply `FluxBoundaryCondition`.
+   _Note_: Do not use `ValueBoundaryCondition` on a wall-normal velocity component.
+   `ImpenetrableBoundaryCondition` is internally enforced for fields created inside the model constructor.
```

Comment:
```suggestion
   _Note_: Do _not_ use `ValueBoundaryCondition` on a wall-normal velocity component. For that, `ImpenetrableBoundaryCondition` is internally enforced for fields created inside the model constructor. The user only needs to change the boundary condition on a wall-normal velocity component if an inflow is desired (this should be done carefully, in order to not violate mass conservation). 
```
Two notes: 

- After the advective forcing PR we need to change this
- Nowhere do we mention `OpenBoundaryConditoin` here. I think we should, especially because it appears in the output of many code snippets and it's bound to leave some readers curious.

Reply:
Well, `ImpenetrableBoundaryCondition` is an alias for `OpenBoundaryCondition{nothing}`, and we don't support any other kind of open boundary condition. I also think we might need to rethink how this works soon (eg an "open" boundary should be specified via grid topology, and `Bounded` should be synonymous with impenetrable for velocity fields, similar to how `Periodic` topology is synonymous with periodic boundary condition).

So I think we can maybe take your comment to mean that we should update `summary(::OpenBoundaryCondition{nothing})` so that it displays as "ImpenetrableBoundaryCondition"?

Reply:
I didn't know we didn't support `OpenBoundaryCondition`. I modified my suggestion to fit that, so now it only mentions `AdvectiveForcing`. 

May I ask what's missing to support it? I've tried it in a couple of toy models and it seemed to work as intended...

> So I think we can maybe take your comment to mean that we should update summary(::OpenBoundaryCondition{nothing}) so that it displays as "ImpenetrableBoundaryCondition"?

I think that would help, but I don't feel strongly about it. I only mentioned it right now because, as it is, `OpenBoundaryCondition` appears on the docs section about boundary conditions but isn't explained anywhere there.

Reply:
It only works for second order advection. The pressure solver isn't right if the velocity is non-constant in time. If it's constant, it doesn't provide any features additional to `PeriodicBoundaryCondition` with a sponge layer. We're discontinuing this boundary condition imminently; but hopefully we will support proper support for open boundary conditions eventually, for sure.

Reply:
there's a comment a few lines down that says:

> Note however that impenetrability is internally enforced for model velocity components,
so that `ImpenetrableBoundaryCondition` is only used for _additional_ velocity components
that are not evolved by a model, such as a velocity component used for (`AdvectiveForcing`)[@ref].

Ill try to combine your suggestion with this.